### DISCUSSION TITLE
SeekToCurrentErrorHandler on KafkaListener

### DISCUSSION THREAD

**Discussion by clydzik** [spring-projects/spring-kafka#1802] [Q&A]
Hello.
I know how to specify errorHandler on ConcurrentKafkaListenerContainerFactory.

But i would like to avoid this and stick with the 'default'.
What i would like to change is to specify my own BackOff on SeekToCurrentErrorHandler. The default is 10 one by one retry

Now i see there is a way to specify errorHandler on @KafkaListener but it cannot be the SeekToCurrentErrorHandler.

What is the simples way to change it for default factory ?

Thanks in Advance
Chris

**Comment by clydzik** [comment]
After some digging...
Well it seems to be pretty straightforward what is a surprise
All need to be done is to expose a bean SeekToCurrentErrorHandler
As stated here:
https://stackoverflow.com/questions/60550354/i-want-to-enable-kafka-errorhandling-by-annotation

Now this will obviously not work if we want to have different backoff for each KafkaListener.
But in that case it may be reasonable to expose a KafkaListenerContainerFactory for each...

**Reply by artembilan** [reply]
That's correct. For now this is only the way, but still: the `ErrorHandler` like this `SeekToCurrentErrorHandler` is really fully tied with the container it is configured for. And that one comes from the mentioned `KafkaListenerContainerFactory`. Therefore it is fully natural to think about different container factories if we would like to have different `ErrorHandler` s.

**Reply by garyrussell** [reply]
You can add a container customizer to the factory to add a different error handler for each listener

```java
@Component
class FactoryCustomizer {

	FactoryCustomizer(ConcurrentKafkaListenerContainerFactory<String, String> factory) {

		factory.setContainerCustomizer(container -> {
			if (container.getGroupId().equals("group1")) {
				container.setErrorHandler(...);
			}
		});
	}

}
```
